src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,230 @@
+import { ExecRenderBaseCommand } from "../../rendering/react/ExecRenderBaseCommand.js";
+import {
+  appInstallationArgs,
+  withAppInstallationId,
+} from "../../lib/app/flags.js";
+import { projectFlags } from "../../lib/project/flags.js";
+import { Flags, ux } from "@oclif/core";
+import React, { ReactNode } from "react";
+import { Text } from "ink";
+import {
+  getAppFromUuid,
+  getAppInstallationFromUuid,
+  getAppVersionFromUuid,
+} from "../../lib/app/uuid.js";
+import {
+  getAllUpgradeCandidatesFromAppInstallationId,
+  getAvailableTargetAppVersionFromExternalVersion,
+  getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates,
+} from "../../lib/app/versions.js";
+import {
+  makeProcessRenderer,
+  ProcessFlags,
+  processFlags,
+} from "../../rendering/process/process_flags.js";
+import { Success } from "../../rendering/react/components/Success.js";
+import { ProcessRenderer } from "../../rendering/process/process.js";
+import { MittwaldAPIV2, MittwaldAPIV2Client } from "@mittwald/api-client";
+import { waitUntilAppIsUpgraded } from "../../lib/app/wait.js";
+import { assertStatus } from "@mittwald/api-client-commons";
+
+type AppApp = MittwaldAPIV2.Components.Schemas.AppApp;
+type AppAppInstallation = MittwaldAPIV2.Components.Schemas.AppAppInstallation;
+type AppAppVersion = MittwaldAPIV2.Components.Schemas.AppAppVersion;
+
+export class UpgradeApp extends ExecRenderBaseCommand<typeof UpgradeApp, void> {
+  static description = "Upgrade target appinstallation to target version";
+  static args = {
+    ...appInstallationArgs,
+  };
+  static flags = {
+    "target-version": Flags.string({
+      description: "Target version to upgrade target app to.",
+    }),
```

Comment:
This this flag is neither required nor has a default value, the documentation should explain what happens when the flag is omitted.

```suggestion
    "target-version": Flags.string({
      description: "target version to upgrade app to; if omitted, target version will be prompted interactively",
    }),
```